src/passes/OptimizeInstructions.cpp

```diff
@@ -3870,7 +3866,22 @@ struct OptimizeInstructions
       }
     }
     {
-      // fval(C) / -x   ==>  -C / x
+      // NaN - x   ==>   NaN'
+      // NaN / x   ==>   NaN'
+      Const* c;
+      if ((matches(curr, binary(Sub, fval(&c), pure())) ||
+           matches(curr, binary(DivS, fval(&c), pure()))) &&
```

Comment:
Is there a reason to match `sub` and `div` specifically? I think this is true for any binary operation except for copysign?

Reply:
It's already handled in https://github.com/WebAssembly/binaryen/pull/4985